Source/CombatExtended/CombatExtended/ArmorUtilityCE.cs

```diff
@@ -101,30 +101,84 @@ public static DamageInfo GetAfterArmorDamage(DamageInfo originalDinfo, Pawn pawn
+                        }
+                        // Hit was deflected, convert damage type
+                        dinfo = GetDeflectDamageInfo(dinfo, hitPart, ref dmgAmount, ref penAmount);
+                        //Check if converted blunt damage could penetrate the shield
+                        if (!TryPenetrateArmor(dinfo.Def, shield.GetStatValue(dinfo.Def.armorCategory.armorRatingStat), ref penAmount, ref dmgAmount))
+                        {
+                            shieldAbsorbed = true;
+                            armorDeflected = true;
+                            dinfo.SetAmount(0);
+                            // Apply secondary damage to shield
+                            if (dinfo.Weapon?.projectile is ProjectilePropertiesCE props && !props.secondaryDamage.NullOrEmpty())
+                            {
+                                foreach (var sec in props.secondaryDamage)
+                                {
+                                    if (shield.Destroyed)
+                                    {
+                                        break;
+                                    }
+                                    if (!Rand.Chance(sec.chance))
+                                    {
+                                        continue;
+                                    }
+                                    var secDinfo = sec.GetDinfo();
+                                    var pen = secDinfo.ArmorPenetrationInt; //GetPenetrationValue(originalDinfo);
+                                    var dmg = (float)secDinfo.Amount;
+                                    TryPenetrateArmor(secDinfo.Def, shield.GetStatValue(secDinfo.Def.armorCategory.armorRatingStat), ref pen, ref dmg, shield);
+                                }
+                            }
 
-                        // Apply secondary damage to shield
-                        if (dinfo.Weapon?.projectile is ProjectilePropertiesCE props && !props.secondaryDamage.NullOrEmpty())
+                            return dinfo;
+                        }
+                        //Blunt damage penetrated the shield, apply the damage to left arm
+                        //Could add a check for having weapon equipped, if not, pawns should be able to hold the shield with both arms, increasing their defence
+                        else
                         {
-                            foreach (var sec in props.secondaryDamage)
+                            shieldAbsorbed = true;
+                            //Priority: Left Arm > Right Arm > Left Shoulder
+                            //It seems that losing left shoulder makes shield unequippable, so no need to add a null check (for now)
+                            BodyPartRecord PartToHit = pawn.health.hediffSet.GetNotMissingParts(depth: BodyPartDepth.Outside, tag: BodyPartTagDefOf.ManipulationLimbCore).First(x => x.IsInGroup(CE_BodyPartGroupDefOf.LeftArm) || x.IsInGroup(CE_BodyPartGroupDefOf.RightArm));
```

Comment:
Throws an exception once both arms are missing:
`CombatExtended :: BulletCE impacting thing Marjot<color=#999999FF>, Scientist</color> of def Human added by mod Core.
System.InvalidOperationException: Sequence contains no matching element
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable\``1[T] source, System.Func\``2[T,TResult] predicate) [0x00011] in <351e49e2a5bf4fd6beabb458ce2255f3>:0 
  at CombatExtended.ArmorUtilityCE.GetAfterArmorDamage (Verse.DamageInfo originalDinfo, Verse.Pawn pawn, Verse.BodyPartRecord hitPart, System.Boolean& armorDeflected, System.Boolean& armorReduced, System.Boolean& shieldAbsorbed) [0x0042c] in <13b22497386e454eb843af3cf4cb7fd3>:0 
  at CombatExtended.HarmonyCE.Harmony_DamageWorker_AddInjury_ApplyDamageToPart.ArmorReroute (Verse.Pawn pawn, Verse.DamageInfo& dinfo, System.Boolean& deflectedByArmor, System.Boolean& diminishedByArmor) [0x0000e] in <13b22497386e454eb843af3cf4cb7fd3>:0 
  at (wrapper dynamic-method) Verse.DamageWorker_AddInjury.Verse.DamageWorker_AddInjury.ApplyDamageToPart_Patch1(Verse.DamageWorker_AddInjury,Verse.DamageInfo,Verse.Pawn,Verse.DamageWorker/DamageResult)
  at Verse.DamageWorker_AddInjury.ApplyToPawn (Verse.DamageInfo dinfo, Verse.Pawn pawn) [0x00091] in <b3561b1be0664abaaf1e5c4602ec1431>:0 
  at Verse.DamageWorker_AddInjury.Apply (Verse.DamageInfo dinfo, Verse.Thing thing) [0x00013] in <b3561b1be0664abaaf1e5c4602ec1431>:0 
  at (wrapper dynamic-method) Verse.Thing.Verse.Thing.TakeDamage_Patch1(Verse.Thing,Verse.DamageInfo)
  at CombatExtended.BulletCE.Impact (Verse.Thing hitThing) [0x00185] in <13b22497386e454eb843af3cf4cb7fd3>:0 `

Reply:
Thanks. Using FirstOrFallback instead. Will return null when both fails.